src/components/Personalization/in-app-message-actions/actions/displayIframeContent.js

```diff
@@ -28,7 +28,7 @@ const dismissMessage = () =>
   [MESSAGING_CONTAINER_ID, OVERLAY_CONTAINER_ID].forEach(removeElementById);
 export const createIframeClickHandler = (
   interact,
-  navigateToUrl = createRedirect(window)
+  navigateToUrl = redirectWithHistory(window)
```

Comment:
I think it would be better not to add the `redirectWithHistory` function, and instead add a second optional parameter to the `createRedirect` method.  That parameter value could be named `preserveHistory` and default to false.  That way existing functionality is the same, but in this one place you can pass in `true` as the second parameter.

What do you think?

Reply:
Now using `preserveHistory ` parameter set to false by default.